plugins/modules/win_scoop.ps1

```diff
@@ -0,0 +1,234 @@
+    $module.FailJson("Failed to find scoop.ps1, make sure it is added to the PATH")
+  }
+
+  return $scoop_app.Path
+}
+
+function Get-ScoopPackages {
+  param(
+    [Parameter(Mandatory = $true)] [string]$scoop_path
+  )
+
+  $command = Argv-ToString -arguments @("powershell.exe", $scoop_path, "export")
+  $res = Run-Command -Command $command
+  if ($res.rc -ne 0) {
+    $module.Result.command = $command
+    $module.Result.rc = $res.rc
+    $module.Result.stdout = $res.stdout
+    $module.Result.stderr = $res.stderr
+    $module.FailJson("Error checking installed packages")
+  }
+
+  $res.stdout -split "`n" |
+  Select-String '(.*?) \(v:(.*?)\) \[(.*?)\]' |
+  ForEach-Object {
+    [PSCustomObject]@{
+      Package = $_.Matches[0].Groups[1].Value
+      Version = $_.Matches[0].Groups[2].Value
+      Bucket  = $_.Matches[0].Groups[3].Value
+    }
+  }
+}
+
+function Get-InstallScoopPackageArguments {
+  $arguments = [System.Collections.Generic.List[String]]@()
+
+  if ($architecture) {
+    $arguments.Add("--arch")
+    $arguments.Add($architecture)
+  }
+  if ($global) {
+    $arguments.Add("--global")
+  }
+  if ($independent) {
+    $arguments.Add("--independent")
+  }
+  if ($no_cache) {
+    $arguments.Add("--no-cache")
+  }
+  if ($skip_checksum) {
+    $arguments.Add("--skip")
+  }
+
+  return , $arguments
+}
+
+function Install-ScoopPackage {
+  param(
+    [Parameter(Mandatory = $true)] [string]$scoop_path,
+    [Parameter(Mandatory = $true)] [String[]]$packages
+  )
+  $arguments = [System.Collections.Generic.List[String]]@("powershell.exe", $scoop_path, "install")
+  $arguments.AddRange($packages)
+
+  $common_args = Get-InstallScoopPackageArguments
+  $arguments.AddRange($common_args)
+
+  $command = Argv-ToString -arguments $arguments
+
+  if (-not $module.CheckMode) {
+    $res = Run-Command -Command $command
+    $module.Result.rc = $res.rc
+
+    if ($module.Verbosity -gt 1) {
+      $module.Result.stdout = $res.stdout
+    }
+  }
+  $module.Result.changed = $true
+}
+
+function Get-UninstallScoopPackageArguments {
+  $arguments = [System.Collections.Generic.List[String]]@()
+
+  if ($independent) {
+    $arguments.Add("--global")
+  }
+  if ($purge) {
+    $arguments.Add("--purge")
+  }
+
+  return , $arguments
+}
+
+function Uninstall-ScoopPackage {
+  param(
+    [Parameter(Mandatory = $true)] [string]$scoop_path,
+    [Parameter(Mandatory = $true)] [String[]]$packages
+  )
+
+  $arguments = [System.Collections.Generic.List[String]]@("powershell.exe", $scoop_path, "uninstall")
+  $arguments.AddRange($packages)
+
+  $common_args = Get-UninstallScoopPackageArguments
+  $arguments.AddRange($common_args)
+
+  $command = Argv-ToString -arguments $arguments
+
+  if (-not $module.CheckMode) {
+    $res = Run-Command -Command $command
+    $module.Result.rc = $res.rc
+
+    if ($module.Verbosity -gt 1) {
+      $module.Result.stdout = $res.stdout
+    }
+    if (-not ($res.stdout -match "ERROR '(.*?)' isn't installed.")) {
```

Comment:
Will this be a problem with other locales?

Reply:
Scoop isn't localised. See [here](https://github.com/lukesampson/scoop/blob/08af9ff6e7f7b017701c0c9114294b13e1e83fb8/lib/core.ps1#L670-L672) for this specific error message.